packages/flutter/lib/src/widgets/form.dart

```diff
@@ -430,8 +434,9 @@ class FormFieldState<T> extends State<FormField<T>> {
 
   @override
   Widget build(BuildContext context) {
-    // Only autovalidate if the widget is also enabled
-    if (widget.autovalidate && widget.enabled)
+    // Only autovalidate if the widget is also enabled and
+    // if its content has changed or has focus
+    if (widget.autovalidate && widget.enabled && (_valueChanged || Focus.isAt(context)))
```

Comment:
Can you remind me why do we need Focus.isAt(context)?

Reply:
I can't tell you right now. Le me investigate.

Reply:
@chunhtai I added this to make this test pass : https://github.com/flutter/flutter/blob/749a30b4380761d327a70df99fc18f2d9f68c99c/packages/flutter/test/material/dropdown_form_field_test.dart#L151

It was to assert that the validation of `DropdownButtonFormField` only happens when it is focused since there the user does not type the text right in the Field.

It seems unnecessary to me right now, since I can just use the value state changes property